cmd/retention-main.go

```diff
@@ -93,6 +109,15 @@ func setRetention(urlStr string, mode *minio.RetentionMode, validity *uint, unit
 	}
 
 	alias, _, _ := mustExpandAlias(urlStr)
+	if !isRecursive {
```

Comment:
should we skip this check here and instead let the server respond with an object does not exist error? This would eliminate the need to have encryption key

Reply:
yes, fixed this